### PR TITLE
Gigantic package refactor

### PR DESCRIPTION
**Pull request by dmacthedestroyer**
This is the result of investigation on PR #29.

This is a **MAJOR BREAKING** change and care should be taken to determine if we want to merge it or not. This will break every dependency that isn't tied to a version in every repository.

I do believe this is an improved layout and I think we should be more careful about where to put things in the future.

**Outline of Changes**
* moved `op` -> `collections`
* moved `gw` -> `gateway`
* broke out the `gorm` functionality between `collections` and `gateway` to better delineate their responsibilities (refactoring identified cyclic dependencies between the two -- now `gateway` depends on `collections` but `collections` is `independent`)
* moved `middleware/auth` to a top-level package and renamed its gRPC interceptor to match the conventions in the [grpc-middleware](https://github.com/grpc-ecosystem/go-grpc-middleware) project
* moved gateway middleware into `gateway` package and renamed its interceptor as above
* removed the `pkg` folder and relocated the sole `gateway.go` file into the `gateway` package

### CODE REVIEW

**Comment by ZachEddy** [comment]
[Here's](https://gist.github.com/ZachEddy/344b3a9fd33f91313f045e3fe61fde25) the tree structure of the new layout.

**Comment by dmacthedestroyer** [comment]
Basically anyone who has imported the toolkit without referencing an explicit version will likely have breaking changes from this PR.

That being said, we are at version 0 and within our right to do so :)

What's the right way to document this?

**Comment by johnbelamaric** [comment]
When we release, we just need to provide explicit instructions for what needs to be done in any client code:

 - regen all generated code
 - fix imports as needed

**Comment by johnbelamaric** [comment]
We should explicitly say change import xyz to import abc, etc.

**Comment by kumaya** [comment]
We will need changes in cli/atlas/templates/* files as well.

**Comment by ZachEddy** [comment]
@Calebjh This also affects protoc-gen-gorm: see [this](https://github.com/infobloxopen/protoc-gen-gorm/blob/d0d10c0518eaf0436d544e73b6938c9f277f6c71/plugin/imports.go#L52).

**Comment by amaskalenka** [comment]
Put my 2 cents in:

The `cli` should be renamed to `cmd`.
The `collections` to `query` and `health` to `k8s`.

The `gorm` should have its own package.

**Comment by dmacthedestroyer** [comment]
would someone please approve this PR?

File: README.md [hunk 1]
```diff
@@ -728,7 +728,7 @@ Note: if you decide to use toolkit provided `infoblox.api.Filtering` proto type,
 ##### How to define filtering in my request?
 
 ```proto
-import "github.com/infobloxopen/atlas-app-toolkit/op/collection_operators.proto";
+import collections;
```
[the remark this answers is not in the document]
Comment (dmacthedestroyer) [reply README.md]: good catch, Andrei!

**Pull request merged**